cube/src/main/java/zone/cogni/asquare/cube/sort/SkolemizedModel.java

```diff
@@ -0,0 +1,48 @@
+package zone.cogni.asquare.cube.sort;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.rdf.model.RDFNode;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.rdf.model.ResourceFactory;
+import org.apache.jena.rdf.model.Statement;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.function.Function;
+
+public class SkolemizedModel implements Function<Model, Model> {
+
+  public SkolemizedModel(@Nonnull String skolemBaseUri) {
+    this.skolemBaseUri = skolemBaseUri;
+  }
+
+  private String skolemBaseUri;
+
+  @Override
+  public Model apply(Model model) {
+    Model result = ModelFactory.createDefaultModel();
+    List<Statement> statements = new StatementSorter().apply(model);
+
+    statements.forEach(statement -> {
+      Resource newSubject = statement.getSubject().isURIResource() ? statement.getSubject()
+                                                                   : getSkolemUri(statement.getSubject());
+
+      RDFNode newObject = !statement.getObject().isAnon() ? statement.getObject()
```

Comment:
I remember you not being a fan of ! operator if not needed. I guess it makes sense to align with other check, so yeah, it's the same for me :)

Reply:
yes. neither am I but else the expression would be much more complex. we need then to check for URI resource or literal.